Issue1 initial tests checkign for failure

apupier opened the pull request.

bfitzpat committed:
issue #1 - adding initial test

-- moving typescript dependency from devDependencies to dependencies to
fix weird issue with vsc prepublish task
-- adjusting travis config
-- fixing versions to build vsc package without errors
-- correcting url in package file

Signed-off-by: Brian Fitzpatrick <bfitzpat@redhat.com>

bfitzpat commented:
@apupier any thoughts on why this is failing now?

bfitzpat commented:
Looks like it may come down to a dependencies issue with typescript? https://github.com/Microsoft/vscode/issues/54332#issuecomment-408793970 - trying that now

apupier committed:
Add missing semicolon #1

Signed-off-by: Aurélien Pupier <apupier@redhat.com>

apupier committed:
Check for command registration #1

now test is failing if a command is returning an error when trying to
execute it (for instance if command is not registered)

Signed-off-by: Aurélien Pupier <apupier@redhat.com>

apupier commented:
> I am not sure we need the lint or gulp bits, but I got it to work in a manner similar to what we did with the atlasmap tests.

they are here to report code quality issues as far as I know. So not necessary fully in this PR scope but is great in general.

bfitzpat commented on package.json:
```diff
@@ -6,20 +6,21 @@
   "repository": {
     "type": "git",
-    "url": "https://github.com/SamVerschueren/vscode-yo"
+    "url": "https://github.com/camel-tooling/vscode-yeoman"
```
Ok. So don't worry about the abandoned original code or provide a link back to it other than documentation?

apupier replied:
I think that just a link inside the documentation is fine.
The provided modifications are great. I was just meaning that if the removal of SamVerschueren references are done in a separated PR, it would make it easier to review. We will have 2 smaller PR concentrated on 2 different goals. Instead of 1 PR with 2 different goals.
As you already created the PR and that is not a very complicated one, i'm fine having both. Just trying to point out easier way to manage PR for the future for similar cases.

bfitzpat replied:
Cool. Will try and keep things more atomic going forward

bfitzpat commented on src/test/command.test.ts:
```diff
@@ -0,0 +1,39 @@
+
+	describe("Yeoman and Yo", () => {
+		before(() => {
+			inputStub.onFirstCall().returns("goonies");
```
This was an initial stab at a test. Yes, we can likely ignore the mock for now and just include it later if we decide to go that route. I'm guessing we can use the Yeoman generator itself that is installed with Yeoman to test the extension, which would be a live test and not require a mock at all.

bfitzpat commented:
@apupier Updated test (thanks for the suggested PR - I took the code and tweaked it a little) and fixed spacing issue.

apupier commented on .vscode/tasks.json:
```diff
@@ -1,8 +1,17 @@
-	"problemMatcher": "$tsc-watch"
+	"tasks": [
+        {
+            "label": "build",
```
space instead of tabs

apupier commented on package.json:
```diff
@@ -46,28 +48,44 @@
+    "gulp-tslint": "^8.1.3",
+    "mocha": "^5.2.0",
+    "sinon": "^7.2.3",
+    "sinon-chai": "^3.3.0",
```
I think the sinon-chai are no more used then but as it is a devdependencies and we will surely use it after that's not a big deal.
Not having the boilerplate code for mock which was not really used in the tests makes the review far easier to review.

bfitzpat replied:
Ok we'll leave it for now. Fixed the spaces issue
